src/OpenColorIO/transforms/builtins/ColorMatrixHelpers.cpp

```diff
@@ -0,0 +1,260 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright Contributors to the OpenColorIO Project.
+
+
+#include <OpenColorIO/OpenColorIO.h>
+
+#include "transforms/builtins/ColorMatrixHelpers.h"
+
+
+namespace OCIO_NAMESPACE
+{
+
+namespace CIE_XYZ_D65
+{
+static const Chromaticities red_xy(1.,     0.    );
+static const Chromaticities grn_xy(0.,     1.    );
+static const Chromaticities blu_xy(0.,     0.    );
+static const Chromaticities wht_xy(0.3127, 0.3290);
+
+const Primaries primaries(red_xy, grn_xy, blu_xy, wht_xy);
+}
+
+// ACES Primaries from SMPTE ST2065-1.
+namespace ACES_AP0
+{
+static const Chromaticities red_xy(0.7347,  0.2653 );
+static const Chromaticities grn_xy(0.0000,  1.0000 );
+static const Chromaticities blu_xy(0.0001, -0.0770 );
+static const Chromaticities wht_xy(0.32168, 0.33767);
+
+const Primaries primaries(red_xy, grn_xy, blu_xy, wht_xy);
+}
+
+namespace ACES_AP1
+{
+static const Chromaticities red_xy(0.713,   0.293  );
+static const Chromaticities grn_xy(0.165,   0.830  );
+static const Chromaticities blu_xy(0.128,   0.044  );
+static const Chromaticities wht_xy(0.32168, 0.33767);
+
+const Primaries primaries(red_xy, grn_xy, blu_xy, wht_xy);
+}
+
+namespace REC709
+{
+static const Chromaticities red_xy(0.64,   0.33  );
+static const Chromaticities grn_xy(0.30,   0.60  );
+static const Chromaticities blu_xy(0.15,   0.06  );
+static const Chromaticities wht_xy(0.3127, 0.3290);
+
+const Primaries primaries(red_xy, grn_xy, blu_xy, wht_xy);
+}
+
+namespace P3_DCI
+{
+static const Chromaticities red_xy(0.680, 0.320);
+static const Chromaticities grn_xy(0.265, 0.690);
+static const Chromaticities blu_xy(0.150, 0.060);
+static const Chromaticities wht_xy(0.314, 0.351);
+
+const Primaries primaries(red_xy, grn_xy, blu_xy, wht_xy);
+}
+
+namespace P3_D65
+{
+static const Chromaticities red_xy(0.680,  0.320 );
+static const Chromaticities grn_xy(0.265,  0.690 );
+static const Chromaticities blu_xy(0.150,  0.060 );
+static const Chromaticities wht_xy(0.3127, 0.3290);
+
+const Primaries primaries(red_xy, grn_xy, blu_xy, wht_xy);
+}
+
+
+// Here are some notes on how one derives the color space conversion
```

Comment:
Great explanation!

Reply:
@doug-walker did an amazing work on the algorithms & explanations.

Reply:
Thanks @doug-walker !